lib/matplotlib/text.py

```diff
@@ -294,9 +294,13 @@ def _get_layout(self, renderer):
         of a rotated text when necessary.
         """
         key = self._get_layout_cache_key(renderer=renderer)
-        if key in self._cached:
-            return self._cached[key]
+        return self._get_layout_cache(key, renderer)
 
+    @functools.lru_cache(maxsize=50)
+    def _get_layout_cache(self, key, renderer):
```

Comment:
This is broken from what @anntzer mentioned, because it's caching on `self`.

Reply:
🐑 Doh! Should have read that closer. I'll move to draft for now and wait until #22271 goes in to have another look at this.

Reply:
FYI #22271 is in.